pppd/main.c

```diff
@@ -1414,7 +1418,7 @@ hup(int sig)
 	kill_my_pg(sig);
     notify(sigreceived, sig);
     if (waiting)
-	write(sigpipe[1], &sig, sizeof(sig));
+	(void) (write(sigpipe[1], &sig, sizeof(sig)) + 1);
```

Comment:
This (void) (... + 1) thing seems like a horrible hack to me. If we can use something like #pragma no_warn_unused or something here that would be better.

The consequence of the write to sigpipe[1] failing is that pppd may not wake up until another event happens. Would it be better to die here if the write fails? It seems like that would just turn a potential hang into a definite crash. Since we can't tell the user why at this point, it seems better to try to limp along. Are there contrary arguments?

Reply:
If we can't work out something using #pragma (and with a comment outlining the reasoning), it's probably better to leave the warning in place than do the (void)(... + 1) hack.